556 demographic viz

tylerwhitson opened the pull request:
Fixes #556 

MVP! Just needs a color scheme and maybe more robust tooltips for the next iteration. Happy to make adjustments.

UPDATE: Made the color scheme a little brighter/easier to differentiate until we get a consistent color scheme.

![demodemo3](https://user-images.githubusercontent.com/35410637/74877531-37051300-532b-11ea-9114-7ea7bfb08c54.gif)

tylerwhitson commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,317 @@
+          }
+          return accumulator;
+        }, 0);
+        const percentage = (categoryTotal / overallTotal) * 100;
```
Good question! In this case I wanted to round to two decimal points because some of the percentages as so small, less than 1% in some cases.

tylerwhitson commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,317 @@
+    { label: "Under 18", categoryValue: [1], color: colors.chartRed },
+    {
+      label: "18 to 44",
+      categoryValue: [2],
```
You're not, thanks for catching that!

tylerwhitson commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,317 @@
+  const getData = categoryValue =>
+    yearsArray().map(year => {
+      if (chartData[year]) {
+        let overallTotal = chartData[year].reduce((accumulator, record) => {
```
Good call! It returns the same value, so I switched to chartData[year].length

tylerwhitson commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,317 @@
+  ];
+
+  // Create array of ints of last 5 years
+  const yearsArray = useCallback(() => {
```
Thanks for the heads up! As we discussed on Slack, I will leave it as-is for now so you can resolve the conflict later. This is good to think about!

mateoclarke commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,312 @@
+
+  return (
+    <Container>
+      <Row style={{ paddingBottom: "0.75em" }}>
```
instead of hardcoding a random fixed value, could you use a bootstrap utility class like `className="pb-3"`
https://getbootstrap.com/docs/4.4/utilities/spacing/

This will ensure consistency over time.

mateoclarke commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,312 @@
+    <Container>
+      <Row style={{ paddingBottom: "0.75em" }}>
+        <Col>
+          <h3 style={{ textAlign: "center" }}>
```
this could be a BS utility class... `className="text-center"`

mateoclarke commented on atd-vzv/src/views/summary/DemographicsByYear.js:
```diff
@@ -0,0 +1,312 @@
+          />
+        </Col>
+      </Row>
+      <Row style={{ paddingTop: "0.75em" }}>
```
same here...

tylerwhitson commented:
> Great work! And solid feedback & discussion with @mddilley!
> 
> 👏👏👏👏👏👏

Thanks, @mateoclarke! I updated all the styling (with the exception of local colors) to Bootstrap classes for this and all the other components for consistency. I also added a couple colors to the color scheme to add a little more contrast and consistency to the chart until we get a global color scheme in place. Merging now!
